themes/default/content/docs/concepts/inputs-outputs/_index.md

```diff
@@ -105,7 +421,7 @@ During some program executions, `apply` doesn’t run. For example, it won’t r
 
 ## All
 
-If you have multiple outputs and need to join them, the `all` function acts like an `apply` over many resources. This function joins over an entire list of outputs. It waits for all of them to become available and then provides them to the supplied callback. This function can be used to compute an entirely new output value, such as by adding or concatenating outputs from two different resources together, or by creating a new data structure that uses them. Just like with `apply`, the result of [Output.all](/docs/reference/pkg/python/pulumi#pulumi.Output.all) is itself an Output<T>.
+If you have multiple outputs and need to use them together, the `all` function acts like an `apply` over many resources, allowing you to use multiple outputs when creating a new output. `all` waits for all output values to become available and then provides them as _plain values_ to the supplied callback. This function can be used to compute an entirely new output value, such as by adding or concatenating outputs from two different resources together, or by creating a new data structure that uses them. Just like with `apply`, the result of [Output.all](/docs/reference/pkg/python/pulumi#pulumi.Output.all) is itself an Output<T>.
```

Comment:
Independent of this PR, but rereading this, there are multiple statements that are not true here. This implies an api that might be nice, but doesn’t exist (something that combines both all and apply into a single call).

Reply:
How about something like:

```
If you need the plain values of multiple outputs at the same time (for example, to create an access policy or a connection string), use the `all` function. `all` takes a list of outputs and returns them as a single output. You can then call `apply` on the result of `all` to access all of the raw values you need in the same context:

// examples here in each language
```

Reply:
I'm going to consider we address this in a further PR